### PR TITLE
[SPARK-14225][SQL] Cap the length of toCommentSafeString at 128 chars

### PR DESCRIPTION
**Pull request by sameeragarwal**
## What changes were proposed in this pull request?

Builds on https://github.com/apache/spark/pull/12022 and (a) appends "..." to truncated comment strings and (b) fixes indentation in lines after the commented strings if they happen to have a `(`, `{`, `)` or `}`
## How was this patch tested?

Manually examined the generated code.

### CODE REVIEW

**Comment by sameeragarwal** [comment]
cc @rxin

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala [hunk 1]
```diff
@@ -49,8 +49,11 @@ private class CodeFormatter {
   private var currentLine = 1
 
   private def addLine(line: String): Unit = {
-    val indentChange =
+    val indentChange = if (line.startsWith("/*") || line.startsWith("//")) {
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala]: this doesn't work if there are multiple lines?

Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala]: btw would be great to add unit tests for this change, once we do it properly

CodeFormatterSuite

Comment (sameeragarwal) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala]: sure, that makes sense. Let me use a small state machine here and add a test.

Comment (sameeragarwal) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala]: I've updated the PR to support basic multi-line comments and added relevant tests. We can probably do fancier things if we'd like to support inline comments (`// hello world /*`) or nested comments, but that's probably not needed here (and should ideally require a standard parser)

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatterSuite.scala [hunk 2]
```diff
@@ -93,4 +100,36 @@ class CodeFormatterSuite extends SparkFunSuite {
...
+    """
+      |/* 001 */ // This is a comment about class A { { { ( (
+      |/* 002 */ class A {
+      |/* 003 */   class body;
+      |/* 004 */ }
+    """.stripMargin
+  }
+
+  testCase("multi-line comments") {
+    """/* This is a comment about
```
Comment (rxin) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatterSuite.scala]: add a space before /*

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala [hunk 3]
```diff
@@ -43,15 +43,41 @@ object CodeFormatter {
...
+  // Tracks the level of indentation in multi-line comment blocks.
+  private var inCommentBlock = false
+  private var indentLevelOutsideCommentBlock = indentLevel
+
   private def addLine(line: String): Unit = {
-    val indentChange =
-      line.count(c => "({".indexOf(c) >= 0) - line.count(c => ")}".indexOf(c) >= 0)
-    val newIndentLevel = math.max(0, indentLevel + indentChange)
+
+    val indentChange = line.count(c => "({".indexOf(c) >= 0) - line.count(c => ")}".indexOf(c) >= 0)
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala]: add some comment saying this is not the most robust implementation but should work for the cases we have for generated code

**Comment by rxin** [comment]
LGTM pending jenkins.

**Comment by rxin** [comment]
I've merged this. Thanks.

**Pull request closed without merging**